xlators/features/locks/src/posix.c

```diff
@@ -1592,11 +1607,18 @@ pl_fgetxattr(call_frame_t *frame, xlator_t *this, fd_t *fd, const char *name,
 {
     int32_t op_ret = 0, op_errno = 0;
     dict_t *dict = NULL;
+    char *client_uid = NULL;
+    pid_t client_pid;
 
     if (!name) {
         goto usual;
     }
 
+    if (frame->root && frame->root->client) {
```

Comment:
If any of these are not present, we shouldn't allow any intrlk command. The function expects these values to be set.